Normalize cloud-controller concept page (#38517)
kubernetes/website: content/en/docs/concepts/architecture/cloud-controller.md

hosts running inside your tenancy with the cloud provider. The node controller performs the following functions:

1. Update a Node object with the corresponding server's unique identifier obtained from the cloud provider API.
1. Annotating and labelling the Node object with cloud-specific information, such as the region the node
   is deployed into and the resources (CPU, memory, etc) that it has available.

> 1. Obtain the node's hostname and network addresses.

Reviewer: We could also:
```suggestion
1. Obtaining the node's hostname and network addresses.
```

`v1/Node`:

- get

> - list

Reviewer 2: Is it important to use lowercase for the method names?

Author: good point, at least this has to be consistent. will revise.

Author: Checked, lowercase is correct.
This PR is good to go as is.

### Service controller {#authorization-service-controller}

> The service controller listens to Service object Create, Update and Delete events and then

Reviewer: Optional further cleanup:
```suggestion
The service controller watches for Service object **create**, **update** and **delete** requests,
and then
```

and an equivalent change everywhere else.

Within the Kubernetes API (eg RBAC types), these verbs are written all-lowercase. See https://kubernetes.io/docs/reference/using-api/api-concepts/ for other precedents.

configures Endpoints for those Services appropriately (for EndpointSlices, the

> kube-controller-manager manages these on demand).

Reviewer: Is this text still true given that we now favor EndpointSlices? I know that one can use Services even without a cloud provider integration.

Author: I think this is still true. cloud-controller-manager is not a must have.

[Cloud Controller Manager Administration](/docs/tasks/administer-cluster/running-cloud-controller/#cloud-controller-manager)
has instructions on running and managing the cloud controller manager.

To upgrade a HA control plane to use the cloud controller manager, see
[Migrate Replicated Control Plane To Use Cloud Controller Manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/).

Want to know how to implement your own cloud controller manager, or extend an existing project?

The cloud controller manager uses Go interfaces to allow implementations from any cloud to be plugged in.
Specifically, it uses the `CloudProvider` interface defined in

> [`cloud.go`](https://github.com/kubernetes/cloud-provider/blob/release-1.26/cloud.go#L43-L69) from

Author: To reviewers: This link is updated from 1.21 to 1.26.
